amlip_cpp/src/cpp/dds/impl/Writer.ipp

```diff
@@ -38,7 +38,13 @@ Writer<T>::Writer(
         topic_,
         qos);
 
-    datawriter_->set_listener(this);
+    auto datawriter_locked = datawriter_.lock_with_exception();
+
+    datawriter_locked->set_listener(this);
```

Comment:
We should change this to avoid known data race, but I guess better in a future PR.